src/features/generic_events/aggregate/index.js

```diff
@@ -121,14 +121,27 @@ export class Aggregate extends AggregateBase {
                 list.getEntries().forEach(entry => {
                   try {
                     handle(SUPPORTABILITY_METRIC_CHANNEL, ['Generic/Performance/' + type + '/Seen'])
+                    const detailObj = agentRef.init.performance.capture_detail ? createDetailAttrs(entry.detail) : {}
                     this.addEvent({
+                      ...detailObj,
                       eventType: 'BrowserPerformance',
                       timestamp: this.toEpoch(entry.startTime),
                       entryName: cleanURL(entry.name),
                       entryDuration: entry.duration,
-                      entryType: type,
-                      ...(entry.detail && { entryDetail: entry.detail })
+                      entryType: type
                     })
+
+                    function createDetailAttrs (obj, rootKey = '', output = {}) {
+                      if (obj === null || obj === undefined) return output
```

Comment:
This is a quite a convoluted method that uses an arg as output but also returns an output. Are you trying to flatten the objected into a single level of key paths and values?

Reply:
the DBC does not flatten a JSON string obj to attributes.  It will report it as a string

This helper function essentially turns
`{foo1: {foo2: {foo3: 'bar3'}, test: 'sup'}`
into
`entryDetail.foo1.foo2.foo3: 'bar3'`
`entryDetail.test: 'sup'`

Reply:
Ok, right. So the way `output` is both an IO arg and a returned val makes this wonky. Arrays are also not considered here so you can end up with `.foo1.0` or `foo1.1`.

Here's a suggested edit:
```js
function flattenJSON(nestedJSON, parentKey = 'entryDetail') {
  let items = {}
  Object.keys(nestedJSON).forEach(key => {
    let newKey = parentKey + '.' + key;
    if (typeof nestedJSON[key] === 'object' && nestedJSON[key] !== null && !Array.isArray(nestedJSON[key])) {
      Object.assign(items, flattenJSON(nestedJSON[key], newKey))
    } else {
      items[newKey] = nestedJSON[key];
    }
  })
  return items
}
```